File: game-box-web-ui/src/app/modules/user/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

import { Observable } from 'rxjs';

import { constants } from '../../../common';
import { IUsersListModel } from '../models/users-list.model';
import { ICreateUserModel } from '../models/create-user.model';

const usersUrl = constants.apiHost + 'users/';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  constructor(private http: HttpClient) { }

  public getUsers$(username: string): Observable<IUsersListModel[]> {
    return this.http.get<IUsersListModel[]>(usersUrl + 'all?username=' + username);
  }

  public lock$(username: string): Observable<string> {
    return this.http.get<string>(usersUrl + 'lock?username=' + username);
  }

  public unlock$(username: string): Observable<string> {
    return this.http.get<string>(usersUrl + 'unlock?username=' + username);
  }

  public addRole$(username: string, role: string): Observable<string> {
    return this.http.get<string>(usersUrl + 'addRole?username=' + username + '&roleName=' + role);
  }

  public removeRole$(username: string, role: string): Observable<string> {
    return this.http.get<string>(usersUrl + 'removeRole?username=' + username + '&roleName=' + role);
  }

  public createUser$(body: ICreateUserModel): Observable<void> {
    return this.http.post<void>(usersUrl + 'create', body);
  }
}

File: Client/GameBox.Admin.UI/Services/UserService.cs
using GameBox.Admin.UI.Model;
using GameBox.Admin.UI.Services.Contracts;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace GameBox.Admin.UI.Services
{
    public class UserService : IUserService
    {
        private readonly string usersUrl;
        private readonly IHttpClientService httpClient;

        public UserService(
            IHttpClientService httpClient,
            ConfigurationSettings config)
        {
            this.httpClient = httpClient;
            this.usersUrl = $"{config.GameBoxApiUrl}users/";
        }

        public async Task<IEnumerable<UserListModel>> GetUsers(string username)
        {
            return await this.httpClient.GetAsync<IEnumerable<UserListModel>>($"{this.usersUrl}all?username={username}");
        }

        public async Task Lock(string username)
        {
            await this.httpClient.GetAsync<string>($"{this.usersUrl}lock?username={username}");
        }

        public async Task Unlock(string username)
        {
            await this.httpClient.GetAsync<string>($"{this.usersUrl}unlock?username={username}");
        }

        public async Task AddRole(string username, string role)
        {
            await this.httpClient.GetAsync<string>($"{this.usersUrl}addRole?username={username}&roleName={role}");
        }

        public async Task RemoveRole(string username, string role)
        {
            await this.httpClient.GetAsync<string>($"{this.usersUrl}removeRole?username={username}&roleName={role}");
        }

        public async Task CreateUser(UserFormModel user)
        {
            await this.httpClient.PostAsync<string>(this.usersUrl + "create", user);
        }
    }
}